File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot/ShooterSubsystems/HoodSubsystem.java
package org.firstinspires.ftc.teamcode.Robot.ShooterSubsystems;
import dev.nextftc.core.subsystems.Subsystem;
import dev.nextftc.hardware.impl.ServoEx;

public class HoodSubsystem implements Subsystem {
    public static HoodSubsystem INSTANCE = new HoodSubsystem();
    public HoodSubsystem() { }

    public ServoEx hoodServo = new ServoEx("hood");

    public void setPosition(double position) {
        hoodServo.setPosition(position);
    }

    @Override
    public void initialize() {
        hoodServo.setPosition(1);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Systems/ShooterSubsystems/HoodSubsystem.kt
package org.firstinspires.ftc.teamcode.Systems.ShooterSubsystems

import dev.nextftc.core.commands.Command
import dev.nextftc.core.subsystems.Subsystem
import dev.nextftc.hardware.impl.ServoEx
import dev.nextftc.hardware.positionable.SetPosition

object HoodSubsystem : Subsystem {
    val hoodServo: ServoEx = ServoEx("hood",-0.1)

    val lowMode: Command = SetPosition(hoodServo, 0.0).requires(this)
    val mediumMode: Command = SetPosition(hoodServo, 0.4).requires(this)
    val highMode: Command = SetPosition(hoodServo, 1.0).requires(this)

    override fun initialize() {
        hoodServo.position = 0.0
    }
}